### PR TITLE
[MI-2458] Added the support for multiple posts for carousel images

### PR DESCRIPTION
**Pull request by manojmalik20**
** Please ignore the CI failing as I'll fix it later **

- Added the logic of adding as many images in a post as possible and then creating new post
- Added the logic of storing the post ids in KV store
- Added the logic of deleting other posts and updating the selected post on clicking
- Added the logic of removing the post action buttons from posts when no action is taken on those posts

### CODE REVIEW

File: server/plugin/virtual_agent.go [hunk 1]
```diff
@@ -512,31 +514,103 @@ func (p *Plugin) CreatePickerAttachment(body *Picker) *model.SlackAttachment {
...
+		p.API.LogDebug("Unable to store the post IDs in KV store", "UserID", userID, "Error", err.Error())
+	}
+
+	for _, postID := range postIDs {
+		go func(postID string) {
+			post, err := p.API.GetPost(postID)
+			if err != nil {
+				p.API.LogDebug("Unable to get the post", "PostID", postID, "Error", err.Error())
+			}
+			if post == nil {
```
Comment (ayusht2810) [inline server/plugin/virtual_agent.go]:
```suggestion
			}
			
			if post == nil {
```

File: server/plugin/virtual_agent.go [hunk 2]
```diff
@@ -512,31 +514,103 @@ func (p *Plugin) CreatePickerAttachment(body *Picker) *model.SlackAttachment {
...
+				return
+			}
+			attachments := post.Attachments()
```
Comment (ayusht2810) [inline server/plugin/virtual_agent.go]:
```suggestion
			}
			
			attachments := post.Attachments()
```

**Review by ayusht2810 — APPROVED** [review]

File: server/plugin/api.go [hunk 3]
```diff
@@ -446,39 +446,83 @@ func (p *Plugin) handlePickerSelection(w http.ResponseWriter, r *http.Request) {
...
+				return
+			}
+
+			if len(postIDs) == 0 {
+				return
+			}
+
+			if err = p.store.StorePostIDs(userID, make([]string, 0)); err != nil {
+				p.API.LogDebug("Unable to store the post IDs in KV store", "UserID", userID, "Error", err.Error())
+			}
```
Comment (raghavaggarwal2308) [inline server/plugin/api.go:464]: What are we doing here?

**Review by raghavaggarwal2308 — APPROVED** [review]

File: server/plugin/api.go (hunk 3 above)
Comment (manojmalik20) [reply server/plugin/api.go:464]: We are clearing the IDs that we stored.

File: server/plugin/virtual_agent.go [hunk 4]
```diff
@@ -454,34 +315,108 @@ func (p *Plugin) CreatePickerAttachment(body *Picker) *model.SlackAttachment {
...
-	return attachments
+	if err := p.store.StorePostIDs(userID, postIDs); err != nil {
+		p.API.LogDebug("Unable to store the post IDs in KV store", "UserID", userID, "Error", err.Error())
+	}
+
+	return nil
+}
+
+func IsCharCountSafe(attachments []*model.SlackAttachment) bool {
+	bytes, _ := json.Marshal(attachments)
```
Comment (Kshitij-Katiyar) [inline server/plugin/virtual_agent.go]: we can at least log this error from marshal in case

**Review by Kshitij-Katiyar — APPROVED** [review]

**Pull request merged**